### ISSUE TITLE
`:conditional-build-up` misses successive conditional`assoc` patterns in `as->`

### ISSUE THREAD

**Issue by jramosg** [clj-kondo/clj-kondo#2846]
**problem**

`:conditional-build-up` warns for repeated `let` rebinding patterns like
`(if pred (assoc m ...) m)`, but it misses the equivalent shape in `as->`.

**repro**
this should also warn
```clojure
(ns repro)

(as-> {:k0 0} m
  (if (p1) (assoc m :k1 1) m)
  (if (p2) (assoc m :k2 2) m))

**Comment by borkdude** [comment]
Not sure how common this is. Is it possible to find examples on github from projects that would use this? Not sure I'm willing to introduce special code for all possible macros, that might be a slippery slope.

**Comment by jramosg** [comment]
If we don’t want to support as->, then I think we should make the boundary explicit in the linter name/docs. Right now `:conditional-build-up` sounds like it catches the pattern generally, while the implementation is specifically about repeated `let` rebinding.

So either:

1. support `clojure.core/as->` as another core way to express the same build-up pattern, or
2. document the linter as only detecting successive `let` rebindings, and maybe rename/reword it accordingly.

My preference is still option 1, with the scope limited to clojure.core forms, not arbitrary macros. But option 2 would at least avoid users expecting equivalent as-> code to be covered.
